### PR TITLE
Route tweaks

### PR DESCRIPTION
**Pull request by thomasmorgan**
these are the commits referenced in issue #241

### CODE REVIEW

File: wallace/custom.py [hunk 1]
```diff
@@ -215,27 +205,27 @@ def node():
                 }
```
Comment (cewing) [inline wallace/custom.py]: The nested try/except in here (line 209/199 in this diff) is another place that would be better served by catching a particular exception type.  Assuming that `exp.known_classes` is a dict-like object, then you should be anticipating a `KeyError`.  

You should internalize the rule of thumb that a *bare except* (using the except clause without specifying one or more exception type to be caught) is a bad idea.

File: wallace/custom.py [hunk 2]
```diff
@@ -420,31 +410,17 @@ def vector():
...
-    try:
-        participant_id = request.values["participant_id"]
-        key = participant_id[0:5]
-    except:
-        exp.log("Error: /vector request, participant_id not specified")
-        page = error_page(error_type="/vector, participant_id not specified")
-        js = dumps({"status": "error", "html": page})
-        return Response(js, status=403, mimetype='application/json')
+    # get the key
+    key = participant_id[0:5]
```
Comment (cewing) [inline wallace/custom.py]: Ahhhhhhhh, deleting code is wonderful!

File: wallace/custom.py [hunk 3]
```diff
@@ -420,31 +410,17 @@ def vector():
...
-            return Response(js, status=403, mimetype='application/json')
-    except:
-        exp.log("Error: /vector request, node_id not specified", key)
-        page = error_page(error_type="/vector, node_id not specified")
+    # check the node_id
+    if not node_id.isdigit():
+        exp.log(
+            "Error: /vector request, non-numeric node_id: {}"
+            .format(node_id), key)
+        page = error_page(error_type="/vector, non-numeric node_id")
```
Comment (cewing) [inline wallace/custom.py]: If it is always an error for node_id to be non-numeric, you can again use flask's routing power to help you here.  For example consider this route (from the flask documentation):

```python
@app.route('/post/<int:post_id>')
```
Here only requests with integer-safe values for post_id will be matched and passed on.  So you can be sure that requests for paths such as `/post/abc` will never match, but `/post/237856` will match.  Additionally, the `post_id` received as an input to the decorated view function will already be cast to an integer, which saves a bit of effort in type conversion.

File: wallace/custom.py [hunk 4]
```diff
@@ -786,44 +748,30 @@ def transmission():
...
-    try:
-        participant_id = request.values["participant_id"]
-        key = participant_id[0:5]
-    except:
-        exp.log("Error: /transmission request, participant_id not specified")
-        page = error_page(error_type="/transmission, participant_id not specified")
-        js = dumps({"status": "error", "html": page})
-        return Response(js, status=403, mimetype='application/json')
+    # get the key
+    key = participant_id[0:5]
```
Comment (cewing) [inline wallace/custom.py]: just an FYI, slicing will always assume the beginning of a string if the `start` part of the slice notation is omitted, so this is the same as saying `participant_id[:5]`

**Comment by cewing** [comment]
@thomasmorgan none of my comments above should prevent you merging these changes.  Assuming you haven't already.  These changes look good to me.  It's wonderful when big chunks of code start disappearing, isn't it?

**Comment by thomasmorgan** [comment]
@cewing thanks for all these points Cris, I'll implement them on this branch too. Also, totally agree, cleaning up code is very cathartic.

Not merging right now as it will stop all experiments working until they are updated.

**Comment by cewing** [comment]
@thomasmorgan I'd be careful about the type of return value that you are generating when your request_parameters function has a problem (https://github.com/suchow/Wallace/blob/route-tweaks/wallace/custom.py#L144).  It's generally a better idea to have a function return a value if it works properly and have it raise an error if it has trouble.  Here, your function returns one type of thing if it works, and returns a different type of thing if it doesn't.  That's less fluent and much harder to test.  

Also, it would appear that the Rogers2b experiment you've sent us depends on this branch of Wallace, as it contains the `/node/<node_id>/infos` route that it used in `tasks.js` in that experiment.  Is that the case?

**Comment by thomasmorgan** [comment]
@cewing Thanks for the tip. Currently it renders and returns an error page if something goes wrong. I thought this would be preferable as otherwise the rendering code will be repeated many times through custom, but I guess returning errors when something goes wrong is worth the cost of repeated code?

And yes, Rogers2b is set up to run on the `route-tweaks` branch of Wallace.

Let me know if you have any other questions.

Cheers

**Comment by cewing** [comment]
@thomasmorgan I would say that there are ways to reduce/eliminate that repeated code, but in the end, yes, I'd say a bit of repetition is a lower price to pay than the price of needing to check what kind of thing is returned by a function in order to determine what to do with it.  That's a style choice, mainly, though.  So feel free to ignore it.

**Comment by cewing** [comment]
@thomasmorgan I am unable to get past the 'ad' endpoint with the Rogers2b experiment.  Is there something special I need to know about how to make that work?

**Comment by thomasmorgan** [comment]
@cewing That's odd. Do you mean you are stuck here:

image

I'm playing it now in debug mode and I get past that point no problem

**Pull request merged**